src/ui/Sort/Sort.ts

```diff
@@ -271,16 +271,47 @@ export class Sort extends Component {
 
   private updateAppearance() {
     $$(this.element).toggleClass('coveo-selected', this.isSelected());
-
     if (this.isToggle()) {
-      var direction = this.currentCriteria ? this.currentCriteria.direction : this.options.sortCriteria[0].direction;
+      const currentDirection = this.currentCriteria ? this.currentCriteria.direction : this.options.sortCriteria[0].direction;
       $$(this.element).removeClass('coveo-ascending');
       $$(this.element).removeClass('coveo-descending');
       if (this.isSelected()) {
-        $$(this.element).addClass(direction === 'ascending' ? 'coveo-ascending' : 'coveo-descending');
+        $$(this.element).addClass(currentDirection === 'ascending' ? 'coveo-ascending' : 'coveo-descending');
       }
     }
   }
+
+  private updateAccessibilityProperties() {
+    this.updateAccessibleSelectedState();
+    this.updateAccessibleLabel();
+  }
+
+  private updateAccessibleSelectedState() {
+    this.element.setAttribute('aria-pressed', this.isSelected().toString());
+  }
+
+  private updateAccessibleLabel() {
+    this.element.setAttribute('aria-label', this.getAccessibleLabel());
+  }
+
+  private getAccessibleLabel() {
+    return this.isToggle() ? this.getAccessibleLabelWithSort() : this.getAccessibleLabelWithoutSort();
+  }
+
+  private getAccessibleLabelWithSort(): string {
+    const localizedCaption = l(this.displayedSortText);
+    const currentDirection = this.currentCriteria ? this.currentCriteria.direction : this.options.sortCriteria[0].direction;
+    let pressingWillSetToAscendingOrder = currentDirection === 'ascending';
```

Comment:
Suggestion:  

```
if (currentDirection === 'ascending' && this.isSelected()) {
  return l('SortResultsByDescending', localizedCaption)
}

if (currentDirection === 'ascending') {
  return l('SortResultsByAscending, localizedCaption)
}

return  return l('SortResultsByDescending', localizedCaption)

```

Seems a bit cleaner to read.

Reply:
Definitely, I'll change it.